src/odemis/driver/xt_client.py

```diff
@@ -2392,3 +2407,21 @@ def _acquire(self):
             logging.exception("Failure in acquisition thread: %s", err)
         finally:
             self._generator = None
+
+
+class XTTKFocus(Focus):
+    """
+    This is an extension of xt_client.Focus class. It overwrites the autofocus function to use the one
+    provided by the flash script.
+    """
+
+    @isasync
+    def applyAutofocus(self):
```

Comment:
`xt_client.Focus.applyAutofocus()` does have a detector argument, so I think it's cleanest if you also have the detector as an argument here.

Reply:
Yes, you're completely right!